sklearn/manifold/t_sne.py

```diff
@@ -746,8 +761,15 @@ def _fit(self, X, skip_num_points=0):
                 print("[t-SNE] Computing {} nearest neighbors...".format(k))
 
             # Find the nearest neighbors for every point
-            knn = NearestNeighbors(algorithm='auto', n_neighbors=k,
-                                   metric=self.metric)
+            if (self.metric_params is None) or not ('p' in self.metric_params):
```

Comment:
I know `p` is just another metric parameter, so it should belong to `metric_params`.
However, for the sake of API homogeneity, I would be favor of adding both parameters `p` and `metric_params`, such as in other estimators. 

(`NearestNeighbors`, `KNeighborsClassifier`, `KNeighborsRegressor`, `RadiusNeighborsClassifier`, `RadiusNeighborsRegressor`, `LocalOutlierFactor`, and `DBSCAN`)

Reply:
Remove redundant parentheses, please